PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1551,6 +1551,13 @@ bool isUnsupportedFileName(const generic_string& fileName)
 				if (pos != std::string::npos)
 					fileNameOnly = fileNameOnly.substr(pos + 1);
 
+				// upperize because the std::find is case sensitive unlike the Windows OS filesystem
+#ifdef UNICODE
+				std::transform(fileNameOnly.begin(), fileNameOnly.end(), fileNameOnly.begin(), ::towupper);
+#else
+				std::transform(fileNameOnly.begin(), fileNameOnly.end(), fileNameOnly.begin(), ::toupper);
+#endif
+

```

Comment:
The UNICODE symbol is always defined in Notepad++ project, and Notepad++ has been a Unicode application since long time ago.
So the 2nd instruction could be removed IMO.

Reply:
Until there are the `generic_string` signatures,(here the `bool isUnsupportedFileName(const generic_string& fileName)` and `bool resolveLinkFile(generic_string& linkFilePath)`), the correct way is to use these UNICODE constructs.

In the future, it will need a some breakpoint PR with refactoring, where the switching to such UNICODE-only chars/strings will be finished. I mean the `generic_string` to `std::wstring`, `TCHAR` to `WCHAR` or `char` to `wchar_t`). And also removing all of these #ifdefs.

But if you rather want to do it in small steps, I will left there only the UNICODE version lines, as requested. (?)

Reply:
> Until there are the generic_string signatures,(here the bool isUnsupportedFileName(const generic_string& fileName) and bool resolveLinkFile(generic_string& linkFilePath)), the correct way is to use these UNICODE constructs.

That's the reason that I use rather `std::wstring` instead of `generic_string`. And in long term, `generic_string` will be removed.

In this case, please change the signature of function resolveLinkFile to `bool resolveLinkFile(std::wstring& linkFilePath)`.

> But if you rather want to do it in small steps, I will left there only the UNICODE version lines, as requested. (?)

Yes, please remove UNICODE - that might lead the confusion that Notepad++ still supports ANSI version.

Reply:
@donho 
Done.

Reply:
@xomx 
Sorry if I wasn't clear.

I saw that you have changed the signature of resolveLinkFile - it's good.
I said "So the 2nd instruction could be removed IMO.", I mean:
instead of 
```cpp
				// upperize because the std::find is case sensitive unlike the Windows OS filesystem
#ifdef UNICODE
				std::transform(fileNameOnly.begin(), fileNameOnly.end(), fileNameOnly.begin(), ::towupper);
#else
				std::transform(fileNameOnly.begin(), fileNameOnly.end(), fileNameOnly.begin(), ::toupper);
#endif
```

use 

```cpp
				// upperize because the std::find is case sensitive unlike the Windows OS filesystem
				std::transform(fileNameOnly.begin(), fileNameOnly.end(), fileNameOnly.begin(), ::towupper);
```

UNICODE is defined in Notepad++ VCPROJ file, TCHAR is always wchar_t - also the generic symbols TCHAR & generic_string will be replaced progressively by wchar_t & wstring respectively.
Since a lot of effort was taken to remove `#ifdef UNICODE` in the past, I don't think it's wise to introduce again such symbol.

Reply:
@donho 
Done.
Sorry, I was not sure before and did not want to bother you with another question…

Reply:
@xomx 
No problem. Look at the bright side, you have pushed me to do the PR #15371 (not finished yet) :)